src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs

```diff
@@ -231,29 +221,35 @@ private void Dispose()
 
         public void Close()
         {
-            if (_disposed)
+            if (Disposed)
                 return;
 
             Close(false);
         }
 
         public void Abort()
         {
-            if (_disposed)
+            if (Disposed)
                 return;
 
             Close(true);
         }
 
         private void Close(bool force)
         {
-            _disposed = true;
-            _context.Connection.Close(force);
+            try
+            {
+                _context.Connection.Close(force);
+            }
+            finally
+            {
+                _responseState = ResponseState.Closed;
```

Comment:
Previously it would register as Disposed before calling Close, now it does it after.  Does that cause any problems?  Is our testing sufficient to catch issues like that?  I worry about the extent of some of these refactorings and whether we're actually able to detect regressions.